app/code/Magento/Inventory/Model/ResourceModel/Reservation/Cleanup.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Inventory\Model\ResourceModel\Reservation;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\Inventory\Setup\Operation\CreateReservationTable;
+use Magento\InventoryApi\Api\Data\ReservationInterface;
+
+/**
+ * The resource model responsible for clearing reservation table (if needed) to prevent overloading, finding complete
+ * pairs of reservations, that is, those (grouped by stock and product) whose sum is 0.
+ *
+ * @package Magento\Inventory\Model\ResourceModel\Reservation
+ */
+class Cleanup
```

Comment:
Maybe we need to provide SPI interface for this action

For example, we have SPI for creation reservations and I could to replace implementation on my own (for example I would like to save reservation in some key-value storage)
But we do not provide possibility to replace implementation for deleting of reservation

It is a little strange for me